src/main/java/com/google/step/snippet/servlets/SearchServlet.java

```diff
@@ -200,17 +207,23 @@ private String getLink(String id, String query) {
     return null;
   }
 
-  private String getPreferredSite(String userId) {
+  private String[] getPreference(String userId) {
+    String[] result = new String[2];
     Query.FilterPredicate filterId =
         new Query.FilterPredicate(ID_PARAMETER, FilterOperator.EQUAL, userId);
     Query userQuery = new Query(USER_PARAMETER).setFilter(filterId);
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     Entity userPreferenceEntity = datastore.prepare(userQuery).asSingleEntity();
-    String site = "";
     if (userPreferenceEntity != null
-        && userPreferenceEntity.getProperty(WEBSITE_PARAMETER) != null) {
-      site = (String) userPreferenceEntity.getProperty(WEBSITE_PARAMETER);
+        && userPreferenceEntity.getProperty(WEBSITE_PARAMETER) != null
+        && userPreferenceEntity.getProperty(LANGUAGE_PARAMETER) != null) {
```

Comment:
It's possible for one but not the other to be null. Can you make these checks separate, so that you only set result[i] if it's not null?